app/Services/BannerFinder.php

```diff
@@ -104,16 +104,16 @@ public static function getBestBanners(array $zones, array $keywords)
         $banners = [];
         foreach ($bannerIds as $bannerId) {
             $banners[] = [
-                'serve_url' => config('app.app_url') . '/img/logo_x.png',
+                'serve_url' => config('app.url') . '/img/logo_x.png',
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "app.url" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=php%3AS1192)